lib/rubocop/cop/infinum/factory_association.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'rubocop'
+
+module RuboCop
+  module Cop
+    module Infinum
+      # This cop looks for 'association' in factories that specify a ':factory' option
+      #
+      # @example
+      #   #bad
+      #    FactoryBot.define do
+      #      factory :book do
+      #        title { 'sth' }
+      #        association :author
+      #      end
+      #    end
+      #
+      #    FactoryBoy.define do
+      #      factory :book do
+      #        title {'sth'}
+      #        author { association :author }
+      #      end
+      #    end
+      #
+      #    #good
+      #    FactoryBot.define do
+      #      factory :book do
+      #        title { 'sth' }
+      #        author { build(:author) }
+      #      end
+      #    end
```

Comment:
Choose a domain for the examples and use meaningful names. The usual convention is blogs and posts, or users and roles, or at the least, foo and bar 🙂 Books are fine too 👍 

```suggestion
      #        title { 'Game of Thrones' }
      #        association :author
      #      end
      #    end
      #
      #    FactoryBoy.define do
      #      factory :book do
      #        title {'Game of Thrones'}
      #        author { association :author }
      #      end
      #    end
      #
      #    #good
      #    FactoryBot.define do
      #      factory :book do
      #        title { 'Game of Thrones' }
      #        author { build(:author) }
      #      end
      #    end
```